ui/src/app/applications/components/application-parameters/source-panel.tsx

```diff
@@ -204,6 +205,11 @@ export const SourcePanel = (props: {
                                                     </div>
                                                 </div>
                                             </div>
+                                            <div className='row argo-form-row'>
+                                                <div className='columns small-10'>
+                                                    <FormField formApi={api} label='Name' field={'spec.source.name'} component={Text}></FormField>
```

Comment:
Should we do an optional param checks, since source and name may not be present?

Reply:
Their values are initialized with an empty string.